tests/stub-cards/valid-cards/node_modules/article-card/schema.js

```diff
@@ -0,0 +1,39 @@
+const JSONAPIFactory = require('@cardstack/test-support/jsonapi-factory');
+module.exports = function({sourceId, packageName}) {
+  // Article Card Schema
+  let factory = new JSONAPIFactory(sourceId, packageName);
+
+  factory.addResource('card-definitions')
+    .withRelated('model', factory.addResource('content-types', 'articles')
+      .withAttributes({ name: 'articles' })
+      .withRelated('fields', [
+        factory.addResource('fields', 'title').withAttributes({
+          name: 'title',
+          'is-metadata': true,
+          'needed-when-embedded': true,
+          'field-type': '@cardstack/core-types::string'
+        }),
+        factory.addResource('fields', 'body').withAttributes({
+          name: 'body',
```

Comment:
seems duplicative to set the resource "id" (it's not the actual ID because the ID is really a composite) and field's "name" to the same value. I think we can add some additional smarts to the `JSONAPIFactory` to add the "name" attribute for the field content type when the `JSONAPIFactory` has knows about the card definition context (because you specified that in it's constructor).

Reply:
Can we just leave the ID blank? `addResource` doesn't require that you pass one.

Reply:
sure